client/src/components/ui/bidder-mytasks-page.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from "react";
+
+import EmptyListDisplay from "./empty-list-display";
+import TaskCard from "./task-card";
+import TopNavtab from "./top-navtab";
+
+interface Task {
+  name: string;
+  title: string;
+  category: string;
+  date: string;
+  location: string;
+  duration: string;
+  estimatePrice: string;
+  myOfferPrice: string;
+  state: "BIDDING" | "EXPIRED" | "ONGOING" | "COMPLETED";
+  priceType: string;
+}
+
+interface TaskListProps {
+  name: string;
+  state: string[];
+}
+
+// using user name and state to filter task
+const filterTasks = (tasks: Task[], name: string, states: string[]) => {
+  return tasks.filter(
+    (task) => task.name === name && states.includes(task.state),
+  );
+};
+
+const TaskList = ({ name, state }: TaskListProps) => {
```

Comment:
Per the style guide, use ``function`` syntax for components, rather than an arrow function.